text/0000-impl-trait-type-aliases.md

```diff
@@ -0,0 +1,372 @@
+- Feature Name: impl_trait_type_aliases
+- Start Date: 2018-08-03
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Allow type aliases to use `impl Trait`, replacing the prototype `existential type` as a way to declare type aliases for opaque, uniquely inferred types.
+
+# Motivation
+[motivation]: #motivation
+
+[RFC 2071](https://github.com/rust-lang/rfcs/blob/master/text/2071-impl-trait-existential-types.md) described a method to define opaque types satisfying certain bounds (described in RFC 2071 and elsewhere as *existential types*). It left open the question of what the precise concrete syntax for the feature should be, opting to use a placeholder syntax, `existential type`. Since then, a clearer picture has emerged as to how to rephrase `impl Trait` in terms of type inference, rather than existentially-quantified types, which also provides new motivation for a proposed concrete syntax making use of the existing and familiar syntax `impl Trait`.
+
+In essence, this RFC proposes that the syntax:
+
+```rust
+type Foo = impl Bar;
+```
+
+be implemented with the same semantics as:
+
+```rust
+existential type Foo: Bar;
+```
+
+and the existing placeholder removed.
```

Comment:
s/and the existing placeholder removed./and the that existing placeholder be removed./